raiden/messages.py

```diff
@@ -337,18 +360,15 @@ def _data_to_sign(self) -> bytes:
         return balance_proof_packed
 
 
-class Processed(SignedMessage):
+class Processed(SignedRetrieableMessage):
     """ All accepted messages should be confirmed by a `Processed` message which echoes the
     orginals Message hash.
-
-    We don't sign `Processed` messages because attack vector can be mitigated and to speed up
-    things.
     """
+    # FIXME: Process should _not_ be SignedRetrieableMessage, but only SignedMessage
```

Comment:
I think `Processed` should be retriable. Imagine Node A sent Node B a `LockExpired` message, which got processed by node B and then the message `Processed` was sent by Node B. Assuming that the `Processed` message was never received by Node A, this will trigger Node A to re-send the lock-expired, which will lead to Node B rejecting that message because it already processed it, but Node A doesn't know that the `LockExpired` message was processed because the `Processed` message was never received. Therefore, i think it should be retried to notify the receiving party.

Reply:
Yes that is a problem we currently have.

It would require that we send `Delivered` messages for `Processed` though (which incidentally @andrevmatos is currently planning to remove).

Reply:
We **do** currently send `Delivered` for `Processed`. #3254 is about removing it.
I think @konradkonrad 's FIXME is exactly about fixing this, as per the talks we had today

Reply:
After the talks we had, I have the feeling the best definition for the `Processed` message is to contain the **nonce** of the latest processed balance proof. For the example that @rakanalh gave Node B would reply to A that everything up to `nonce` `X` has been processed, allowing A to clear the queue

(My previous idea was to just replay with a processed if the `nonce` was lower then the current expected `nonce`, but this is makes it a bit hard to describe the semantics of the message is a few words)